File: onlinepayments-sdk/src/main/java/com/onlinepayments/sdk/client/android/model/validation/ValidationRuleTermsAndConditions.java
/*
 * Copyright 2020 Global Collect Services B.V
 */

package com.onlinepayments.sdk.client.android.model.validation;

import android.util.Log;

import com.onlinepayments.sdk.client.android.model.PaymentRequest;

import java.security.InvalidParameterException;

/**
 * Validation rule for terms and conditions.
 */
public class ValidationRuleTermsAndConditions extends AbstractValidationRule {

    private static final long serialVersionUID = 2209679897444037061L;

    private static final String TAG = ValidationRuleTermsAndConditions.class.getName();

    /**
     * @deprecated This constructor is for internal use only.
     */
    @Deprecated
    public ValidationRuleTermsAndConditions() {
        super("termsAndConditions", ValidationType.TERMSANDCONDITIONS);
    }

    /**
     * @deprecated In a future release, this constructor will be removed.
     */
    @Deprecated
    public ValidationRuleTermsAndConditions(String errorMessage, ValidationType type) {
        super(errorMessage, type);
    }

    /**
     * Validates that the terms and conditions have been accepted.
     *
     * @param paymentRequest the fully filled {@link PaymentRequest} that will be used for doing a payment
     * @param fieldId the ID of the field to which to apply the current validator
     *
     * @return true, if the value in the field with fieldId is true; false, if the value in the field is false or if the fieldId could not be found
     */
    @Override
    public boolean validate(PaymentRequest paymentRequest, String fieldId) {
        if (paymentRequest == null) {
            throw new InvalidParameterException("Error validating, paymentRequest may not be null");
        }
        if (fieldId == null) {
            throw new InvalidParameterException("Error validating, fieldId may not be null");
        }

        String value = paymentRequest.getValue(fieldId);
        return Boolean.parseBoolean(value);
    }

    /**
     * @return <code>True</code>
     * @deprecated use {@link #validate(PaymentRequest, String)} instead
     */
    @Override
    @Deprecated
    public boolean validate(String text) {
        Log.w(TAG, "This method is deprecated and should not be used! Use <validate(PaymentRequest paymentRequest, String)> instead.");
        return true;
    }
}

File: onlinepayments-sdk/src/main/kotlin/com/onlinepayments/sdk/client/android/model/validation/ValidationRuleTermsAndConditions.kt
/*
 * Do not remove or alter the notices in this preamble.
 *
 * Copyright © 2025 Worldline and/or its affiliates.
 *
 * All rights reserved. License grant and user rights and obligations according to the applicable license agreement.
 *
 * Please contact Worldline for questions regarding license and user rights.
 */
package com.onlinepayments.sdk.client.android.model.validation

import com.onlinepayments.sdk.client.android.model.PaymentRequest

/**
 * Validation rule for terms and conditions.
 */
class ValidationRuleTermsAndConditions internal constructor(): AbstractValidationRule("termsAndConditions", ValidationType.TERMSANDCONDITIONS) {

    /**
     * Validates that the terms and conditions have been accepted.
     *
     * @param paymentRequest the fully filled [PaymentRequest] that will be used for doing a payment
     * @param fieldId the ID of the field to which to apply the current validator
     *
     * @return true, if the value in the field with fieldId is true; false, if the value in the field is false or if the fieldId could not be found
     */
    override fun validate(paymentRequest: PaymentRequest, fieldId: String): Boolean {
        return paymentRequest.getValue(fieldId).toBoolean()
    }

    companion object {
        @Suppress("Unused")
        private val serialVersionUID = 2209679897444037061L
    }
}
